components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/utils/APIUtil.java

```diff
@@ -902,6 +902,136 @@ public static API getAPI(GovernanceArtifact artifact)
         return api;
     }
 
+    /**
+     * This method is used to get an API in the Light Weight manner.
+     * @param artifact  generic artfact
+     * @return this will return an API for the selected artifact.
+     * @throws APIManagementException , if invalid json config for the API or Api cannot be retrieved from the artifact
+     */
+    public static API getLightWeightAPI(GovernanceArtifact artifact)
+            throws APIManagementException {
+
+        API api;
+        try {
+            String providerName = artifact.getAttribute(APIConstants.API_OVERVIEW_PROVIDER);
+            String apiName = artifact.getAttribute(APIConstants.API_OVERVIEW_NAME);
+            String apiVersion = artifact.getAttribute(APIConstants.API_OVERVIEW_VERSION);
+            APIIdentifier apiIdentifier = new APIIdentifier(providerName, apiName, apiVersion);
+            api = new API(apiIdentifier);
+            int apiId = ApiMgtDAO.getInstance().getAPIID(apiIdentifier, null);
+            if (apiId == -1) {
```

Comment:
Can the apiId be -1 while an apiIdentifier exists? Isn't this a scenario where we should throw an exception?